builtin/logical/pki/acme_state.go

```diff
@@ -128,17 +134,37 @@ type acmeAccount struct {
 }
 
 func (a *acmeState) CreateAccount(ac *acmeContext, c *jwsCtx, contact []string, termsOfServiceAgreed bool) (*acmeAccount, error) {
+	// Write out the thumbprint value/entry out first, if we get an error mid-way through
+	// this is easier to recover from if we have an entry in this table with no corresponding
+	// kid entry, as the end-user will most likely retry with the same key but will have a
+	// newly generated kid value.
```

Comment:
I think the implication here needs to be stated: that the new kid value would cause a rewrite of both the thumbprint and the account entry, which'd clobber the dangling, incorrect kid reference here. Otherwise, I was going to suggest the other order until I realized this approach would be self-healing _if_ they re-attempted the account creation. ;-)